feature/bgp/routereflector/route-reflector-test

```diff
@@ -0,0 +1,45 @@
+Summary: 
+ - BGP route reflector capabilities check
+ - Ensure functionality of different OC paths for "supported-capabilities", "BGP peer-type", "BGP Neighbor details" and "BGP transport session parameters"
+
+Topology:
+ATE (Port1) <-EBGP-> (Port1) DUT (Port2) <-IS-IS/IBGP-> (Port2) ATE, DUT (Port3) <-IS-IS/IBGP-> (Port3) ATE
+  - Connect ATE Port1 to DUT port1 (EBGP peering)
+  - Connect ATE Port2 to DUT port2 (For IS-IS adjacency and IBGP peer reachaility)
+  - Connect ATE Port3 to DUT port3 (For IS-IS adjacency and IBGP peer reachaility)
+
+Procedure:
+  - Establish IS-IS adjacency between ATE Port2 <-> DUT Port2, ATE Port3 <-> DUT Port3. 
+  - Establish BGP sessions as follows between ATE and DUT . 
+    - ATE port 2 and ATE port3 are emulating RR clients peered with the DUT acting as the RR server. DUT's loopback address should be used for the IBGP peering and 
+      "transport/config/local-address" OC path should be used on DUT to configure BGP transport address for IBGP peering. ATE addresses used for the IBGP peering (different from ATE Port1 and ATE Port2 addreses) and DUT 
+      loopback addresses should be reachable via IS-IS. Each of the IBGP peering should advertise 5M IPv4 and 1M IPv6 prefixes.
```

Comment:
It would be helpful to give additional guidance about these routes, e.g. how many of them are the same prefix vs unique prefixes from each RRC? Basic RR reflection would cause each RRC to see all prefixes, so the validation could look for reflected unique (i.e. otherwise unknown to the RR) prefixes on each RRC.

Reply:
Added more text to clarify

Reply:
This is to address b/299390379 which is RR @ scale. So this test should cover for that requirement. I have updated the test naming to also reflect the same.

Reply:
@sachendras, I think you still  need to specify if the two iBGP sessions should contain the same 5+1M prefixes or if they are expected to be unique?  I am guessing the two sessions are expected to contain the exact same 5+1M prefixes? If yes, can you state that?

Reply:
I thought I clarified: Here's what I had for local prefixes


- Among the 2M IPv4 prefixes as well, 1.5M are non RFC1918 or RFC6598 addresses and represent Internet prefixes. (adding new for clarity) These prefixes should be common between the 2xDUTs with different path-attributes for protocol next-hop, AS-Path and community
- Balance 500k IPv4 and 400k IPv6 prefixes represent internal subnets and should include some prefixes that are unique to each of the ATEs. (adding new for more clarity) Remaining prefixes in the mix need to be common between the 2xATEs and should have identical path attributes except for the protocol next-hops.

Reply:
Got it, I think I reviewed an earlier commit that didn't include this text.  I can understand it now, thanks!